src/client/baseClient.ts

```diff
@@ -47,8 +57,12 @@ export class BaseClient {
         config,
         tooling
       );
+    } else if (this.isDpopCredentials(config.authStrategy)) {
+      this.getAccessToken = config.authStrategy.getAccessToken;
+      this.getDpopHeader = config.authStrategy.getDpopHeader;
     } else {
-      this.getAuthorizationHeader = config.authStrategy;
+      this.getAuthorizationHeader =
+        config.authStrategy as GetAuthorizationHeader;
```

Comment:
If you need to cast here, because Typescript complains. Either your typing are incorrect or you are not handling all the proper cases. Please revisit

```suggestion
        config.authStrategy as GetAuthorizationHeader;
```

Reply:
changed